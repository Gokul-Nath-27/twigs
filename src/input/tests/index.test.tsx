import React from 'react';
import 'jest';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Input } from '../index';

const RightArrow = () => (
  <svg width="12" height="12" viewBox="0 0 6 10" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path d="M1 9L5 5L1 1" stroke="white" strokeWidth="1.5" strokeLinecap="round" strokeLinejoin="round" />
  </svg>
);

describe('Input', () => {
  it('renders without crashing', () => {
    render(<Input />);
  });

  it('matches snapshot: primary with icon on left', () => {
    const tree = renderer.create(
      <Input
        iconLeft={<RightArrow />}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot: secondary with icon on right', () => {
    const tree = renderer.create(
      <Input
        iconRight={<RightArrow />}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot: text with icon on both sides', () => {
    const tree = renderer.create(
      <Input
        iconRight={<RightArrow />}
        iconLeft={<RightArrow />}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
