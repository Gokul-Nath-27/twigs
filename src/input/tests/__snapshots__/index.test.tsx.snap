// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot: primary with icon on left 1`] = `
<div
  className="twigs-c-PJLV twigs-c-PJLV-ijBDafv-css"
  data-testid="box"
>
  <div
    className="twigs-c-PJLV twigs-c-jtoFuQ twigs-c-jtoFuQ-gRAykL-size-md twigs-c-jtoFuQ-igKkDnE-css"
    data-testid="box"
  >
    <svg
      fill="none"
      height="12"
      viewBox="0 0 6 10"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 9L5 5L1 1"
        stroke="white"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="1.5"
      />
    </svg>
  </div>
  <input
    className="twigs-c-fbCaTR twigs-c-fbCaTR-hvIBXR-size-md twigs-c-fbCaTR-ikaYCia-css"
    data-testid="input"
  />
</div>
`;

exports[`Input matches snapshot: secondary with icon on right 1`] = `
<div
  className="twigs-c-PJLV twigs-c-PJLV-ijBDafv-css"
  data-testid="box"
>
  <input
    className="twigs-c-fbCaTR twigs-c-fbCaTR-hvIBXR-size-md twigs-c-fbCaTR-idGgydZ-css"
    data-testid="input"
  />
  <div
    className="twigs-c-PJLV twigs-c-jtoFuQ twigs-c-jtoFuQ-gRAykL-size-md twigs-c-jtoFuQ-ihNJuUr-css"
    data-testid="box"
  >
    <svg
      fill="none"
      height="12"
      viewBox="0 0 6 10"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 9L5 5L1 1"
        stroke="white"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="1.5"
      />
    </svg>
  </div>
</div>
`;

exports[`Input matches snapshot: text with icon on both sides 1`] = `
<div
  className="twigs-c-PJLV twigs-c-PJLV-ijBDafv-css"
  data-testid="box"
>
  <div
    className="twigs-c-PJLV twigs-c-jtoFuQ twigs-c-jtoFuQ-gRAykL-size-md twigs-c-jtoFuQ-igKkDnE-css"
    data-testid="box"
  >
    <svg
      fill="none"
      height="12"
      viewBox="0 0 6 10"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 9L5 5L1 1"
        stroke="white"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="1.5"
      />
    </svg>
  </div>
  <input
    className="twigs-c-fbCaTR twigs-c-fbCaTR-hvIBXR-size-md twigs-c-fbCaTR-iehajXC-css"
    data-testid="input"
  />
  <div
    className="twigs-c-PJLV twigs-c-jtoFuQ twigs-c-jtoFuQ-gRAykL-size-md twigs-c-jtoFuQ-ihNJuUr-css"
    data-testid="box"
  >
    <svg
      fill="none"
      height="12"
      viewBox="0 0 6 10"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 9L5 5L1 1"
        stroke="white"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="1.5"
      />
    </svg>
  </div>
</div>
`;
